File: core/src/main/java/com/canyon/commons/IPAddressUtils.java
package com.canyon.commons;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class IPAddressUtils {
	public static boolean validIP(String strIPAddress) {
		try {
			if (strIPAddress == null || strIPAddress.isEmpty()) {
				return false;
			}

			String[] parts = strIPAddress.split("\\.");
			if (parts.length != 4) {
				return false;
			}

			for (String s : parts) {
				int i = Integer.parseInt(s);
				if ((i < 0) || (i > 255)) {
					return false;
				}
			}
			if (strIPAddress.endsWith(".")) {
				return false;
			}

			return true;
		} catch (NumberFormatException nfe) {
			return false;
		}
	}
	
	public static InetAddress toIPAddress(String strIPAddress){
		InetAddress result = null;
		try {
			result = InetAddress.getByName(strIPAddress);
		} catch (UnknownHostException e) {
		}
		return result;
	}

	public static long toLong(String strIPAddress){
		long result = 0L;
		InetAddress ip = toIPAddress(strIPAddress);
		if(ip != null){
			result = toLong(ip);
		}
		return result;
	}
	
	public static long toLong(InetAddress ip){
		long result = 0L;
		byte[] octets = ip.getAddress();
        for (byte octet : octets) {
            result <<= 8;
            result |= octet & 0xff;
        }
        return result;
	}

	public static boolean contains(String startIP, String endIP, String ip){
        long ipStart  = toLong(startIP);
        long ipEnd  = toLong(endIP);
        long ipTest  = toLong(ip);
        
        return ipTest >= ipStart && ipTest <= ipEnd;
	}
}

File: commons/src/main/kotlin/com/canyon/commons/IPAddressUtils.kt
package com.canyon.commons

import java.net.InetAddress
import kotlin.experimental.and

object IPAddressUtils {
    fun validIPAddress(strIPAddress: String): Boolean {
        if (strIPAddress.isEmpty()) {
            return false
        }

        val parts = strIPAddress.split(".")
        if (parts.size != 4) {
            return false
        }

        for (s in parts) {
            val i = Integer.parseInt(s)
            if (i < 0 || i > 255) {
                return false
            }
        }
        return !strIPAddress.endsWith(".")
    }

    fun toIPAddress(strIPAddress: String): InetAddress {
        return InetAddress.getByName(strIPAddress)
    }

    fun toLong(strIPAddress: String): Long {
        var result = 0L
        val ip = toIPAddress(strIPAddress)
        if (ip != null) {
            result = toLong(ip)
        }
        return result
    }

    fun toLong(ip: InetAddress): Long {
        var result = 0L
        val octets = ip.address
        for (octet in octets) {
            result = result shl 8
            result = result or (octet and 0xff.toByte()).toLong()
        }
        return result
    }

    fun contains(startIP: String, endIP: String, ip: String): Boolean {
        val ipStart = toLong(startIP)
        val ipEnd = toLong(endIP)
        val ipTest = toLong(ip)

        return ipTest in ipStart..ipEnd
    }
}
